torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -185,8 +185,7 @@ Tensor norm_backward(Tensor grad, const Tensor & self, const optional<Scalar> &
 }
 
 Tensor pow_backward(Tensor grad, const Tensor & self, const Scalar & exponent_) {
-  auto exponent = (exponent_.isComplex()) ? exponent_.toComplexDouble() : exponent_.toDouble();
-  if (exponent == 0.0) {
+  if ((exp.isComplex() && (exp.toComplexDouble() == 0.0)) || (!exp.isComplex() && (exp.toDouble() == 0.0))) {
```

Comment:
I feel like there are going to be other places in the codebase where we want to compare a Scalar against a particular value. I'm not sure if there's a good reason why we don't have operator==.

Should we provide something like a helper function `is_scalar_equal(Scalar, double)`?

Reply:
makes sense! added an `equal` method